### PR TITLE
k8s: Use `configmap` for API deployment

### PR DESCRIPTION
**Pull request by JenySadadia**
Kubernetes provides `ConfigMap` mechanism used to configure pods efficiently.

Use `ConfigMap` for providing environment variables to API deployment.

Instead of adding environment variable values for Redis hostname and Mongo DB URL directly into `api` deployment file, reference `ConfigMap` to read values.

- Added manifest file to generate `configmap`
- Updated `api` deployment file to reference configmap variables
- Renamed directory from `hack` to `scripts`
- Added command to generate configmap to `scripts/apply-all.sh`

### CODE REVIEW

**Comment by gctucker** [comment]
@nuclearcat @sbdtu5498 Hello, could you please take a look and review this PR?

**Comment by sbdtu5498** [comment]
Everything seems fine to me. Just a small nit.

File: kube/minkube/README.md [hunk 1]
```diff
@@ -137,15 +137,15 @@ kubectly apply -f api-deployment.yaml
 
 ## Cleaning up everything afterwards
 
-To clean up everything simply use the ```clean-all.sh``` script provided in the root of the kube/minikube/hack folder by running
+To clean up everything simply use the ```clean-all.sh``` script provided in the root of the kube/minikube/scripts folder by running
```
Comment (sbdtu5498) [inline kube/minkube/README.md:140]: @JenySadadia how about calling clean-all.sh tear.sh instead.

**Comment by nuclearcat** [comment]
I guess its need rebase now, as minkube changed to minikube

**Comment by JenySadadia** [comment]
> I guess its need rebase now, as minkube changed to minikube

Rebased it.

File: kube/minkube/README.md (hunk 1 above)
Comment (JenySadadia) [reply kube/minkube/README.md]: `delete-resources.sh` looks more suitable to me.

**Pull request merged**